### PR TITLE
chore(deps): update

### LINKED ISSUE #46: update depndencies

### PR DESCRIPTION
**Pull request by Macavirus**
closes #46

currently failing a couple tests due to typescript upgrade that change how keys are indexed:
see:
https://github.com/microsoft/TypeScript/issues/32704
https://github.com/microsoft/TypeScript/pull/30769

@natp0ng maybe you can have a look since you were just doing a similar upgrade recently

### CODE REVIEW

**Comment by vic08** [comment]
@Macavirus  build is red

**Comment by Macavirus** [comment]
This one still needs some attention to clean up `any` before merging.

File: src/MessageApp.test.ts [hunk 1]
```diff
@@ -81,7 +81,11 @@ test("MessageApp.connect() relays bridge.connect and injects a handler", (done)
 });
 
 test("MessageApp.on()", (done) => {
-    const app = new MessageApp({}, fakeBridge);
+    const app = new MessageApp({
```
Comment (wmathes) [inline src/MessageApp.test.ts]: what's the point of injecting a handler here? :)

File: src/MessageApp.ts [hunk 2]
```diff
@@ -12,8 +12,10 @@ export interface IMessage<Payload = any, Type = any, Meta = any> {
     meta?: Meta;
 }
 
+export type IPayload = (payload: void | any) => void;
```
Comment (wmathes) [inline src/MessageApp.ts]: its not a payload, this should be named `IPayloadHandler`

Comment (wmathes) [reply src/MessageApp.ts]: or simply invite `IMessageHandler`

Comment (wmathes) [reply src/MessageApp.ts]: even better
`export type IMessageHandler<T extends object = any> = (payload: void | T) => void;`

Comment (wmathes) [reply src/MessageApp.ts]: or if you touch it anyway, use `@screencloud/event-emitter`, which offers the same functionality as `.on`/`.off` etc. but fully tested

**Comment by wmathes** [comment]
wtf... why did i get a notification for this? :D it's from january

**Pull request closed without merging**